app.py

```diff
@@ -14,35 +16,80 @@
 app_env = AppEnv()
 aws_rds = app_env.get_service(name="example-website-api-database")
 
-
-connection = psycopg2.connect(
-    host=aws_rds.credentials.get("host"),
-    user=aws_rds.credentials.get("username"),
-    password=aws_rds.credentials.get("password"),
-    database=aws_rds.credentials.get("name"),
-    port=aws_rds.credentials.get("port"),
-)
-
-
+# Set up pool as a function so in the case it dies we can recreate it 
+def setup_connection_pool():
+    
+    database_url = os.getenv("DATABASE_URL")
+    
+    pool = ConnectionPool(
+        conninfo=database_url,
+        min_size=1,
+        max_size=20,
+    )
+    return pool
+
+# Immediately create pool at module import 
+connection_pool = setup_connection_pool()
+
+# Maybe dont need atexit if using context manager per psycopg3 docs
+atexit.register(connection_pool.close)
+
+# Recreate pool if it dies
+def recreate_connection_pool():
+    global connection_pool
+    
+    # Close if it exists
+    if connection_pool is not None:
```

Comment:
The variable will never be `None`, it will be present but not an active connection, more notes on this below